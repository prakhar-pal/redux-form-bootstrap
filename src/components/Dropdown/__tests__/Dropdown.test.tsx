import * as React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Dropdown } from '..';
import { DropdownOption } from "..";

export const options: DropdownOption[] = [
    {value: "You haven't selected a name yet!", type: "disabled"},
    {value: "Karen", type: "header"},
    {type: "divider"},
    {value: "Gillian", type: "header"},
    {type: "divider"},
    {value: "Bucky", type: "header"},
    {type: "divider"},
    {value: "Bikrum", type: "header"},
    {type: "divider"}
]


describe('Dropdown Component', () => {

    it('renders', () => {
        const app = render(
            <Dropdown
                text="Select a name"
                options={options}
                onChange={jest.fn()}
            />);
        expect(app).toMatchSnapshot();
    });

    // it('handles onChange', () => {
    //     const handleChange = jest.fn();
    //     render(
    //         <Dropdown
    //             text="Select a name"
    //             options={options}
    //             onChange={handleChange}
    //         />);
    //     const btnGroupEl = document.querySelector('.btn-group');
    //     btnGroupEl && userEvent.click(btnGroupEl);
    //     expect(handleChange).toHaveBeenCalledTimes(1);
    // });
});