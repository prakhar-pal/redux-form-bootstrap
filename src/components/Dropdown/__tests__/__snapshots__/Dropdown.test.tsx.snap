// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="btn-group"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle btn btn-info btn-sm"
          type="button"
        >
          Select a name
        </button>
        <div
          aria-hidden="true"
          class="dropdown-menu"
          role="menu"
          tabindex="-1"
        >
          <button
            class="disabled dropdown-item"
            disabled=""
            tabindex="-1"
            type="button"
          >
            You haven't selected a name yet!
          </button>
          <h6
            class="dropdown-header"
            tabindex="-1"
          >
            Karen
          </h6>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <h6
            class="dropdown-header"
            tabindex="-1"
          >
            Gillian
          </h6>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <h6
            class="dropdown-header"
            tabindex="-1"
          >
            Bucky
          </h6>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <h6
            class="dropdown-header"
            tabindex="-1"
          >
            Bikrum
          </h6>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="btn-group"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="dropdown-toggle btn btn-info btn-sm"
        type="button"
      >
        Select a name
      </button>
      <div
        aria-hidden="true"
        class="dropdown-menu"
        role="menu"
        tabindex="-1"
      >
        <button
          class="disabled dropdown-item"
          disabled=""
          tabindex="-1"
          type="button"
        >
          You haven't selected a name yet!
        </button>
        <h6
          class="dropdown-header"
          tabindex="-1"
        >
          Karen
        </h6>
        <div
          class="dropdown-divider"
          tabindex="-1"
        />
        <h6
          class="dropdown-header"
          tabindex="-1"
        >
          Gillian
        </h6>
        <div
          class="dropdown-divider"
          tabindex="-1"
        />
        <h6
          class="dropdown-header"
          tabindex="-1"
        >
          Bucky
        </h6>
        <div
          class="dropdown-divider"
          tabindex="-1"
        />
        <h6
          class="dropdown-header"
          tabindex="-1"
        >
          Bikrum
        </h6>
        <div
          class="dropdown-divider"
          tabindex="-1"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
